### ISSUE TITLE
Support for OIDC issuer flag in azurerm_kubernetes_cluster

### ISSUE THREAD

**Issue by kratkyzobak** [hashicorp/terraform-provider-azurerm#14977]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for OIDC issuer (Preview) feature for AKS to be able to configura passwordless POD auth against AzureAD (or any else OIDC RP)

### New or Affected Resource(s)

* azurerm_kubernetes_cluster

### Potential Terraform Configuration

```hcl
resource azurerm_kubernetes_cluster "aks" {
     ..
     oidc_issuer = true # az aks create -n aks -g myResourceGroup --enable-oidc-issuer
     ...
}

resource "azuread_application_federated_identity_credential" "aks-app" {
  application_object_id = azuread_application.app.object_id
  display_name          = "default"
  description           = ""
  audiences             = ["api://oidc-sa-identity"]
  issuer                = azurerm_kubernetes_cluster.aks.oidc_issuer_url # az aks show --resource-group <resource_group> --name <cluster_name> --query "oidcIssuerProfile.issuerUrl" -otsv
  subject               = "system:serviceaccount:NAMESPACE:SERVICEACCOUNT"
}
```

### References
See https://azure.github.io/azure-workload-identity/docs/installation/managed-clusters.html for connecting AKS to WIF

* #0000

**Comment by stijndehaes** [comment]
I wanted to look into building this feature. But the field is only available in the new [SDK](https://github.com/Azure/azure-sdk-for-go/blob/da0fd4d0bcf55c6bd80741101e5892f6e5d582c4/sdk/resourcemanager/containerservice/armcontainerservice/zz_generated_models.go#L1568). I don't know if a migration to that SDK is already planned? Or there is anything blocking there. I am willing to help #13178

**Comment by kratkyzobak** [comment]
Keeping very simple workaround (keep in mind, you have to enable preview per subscription prior to running this)

Enable OIDC issuer after creation:
```hcl
provisioner "local-exec" {
    command = "az aks update -n \"${azurerm_kubernetes_cluster.aks.name}\" -g \"${azurerm_kubernetes_cluster.aks.resource_group_name}\" --subscription \"${data.azurerm_client_config.current.subscription_id}\" --enable-oidc-issuer"
}
```

Fetch OIDC issuer URL:
```hcl
data "external" "aks_oidc_issuer" {
  depends_on = [null_resource.aks_oidc_issuer]

  program = [
    "bash",
    "${path.module}/aks-oidc-issuer-url.sh",
    azurerm_kubernetes_cluster.aks.id
  ]
}
```

Fetch uses this script named `aks-oidc-issuer-url.sh`:
```bash
#!/bin/bash

echo "{\"issuerURL\": \"`az rest --uri "${1}?api-version=2021-11-01-preview" --query properties.oidcIssuerProfile.issuerURL | jq -r`\" }"
```

Then you can use output:
```hcl
output "aks_oidc_issuer_url" {
  value = data.external.aks_oidc_issuer.result.issuerURL
}
```


Has lot of caveats as it has external dependencies (az command with aks-preview, bash and jq), but still leaving here for those, who needs this. I'm using this in my module and waiting to be implemented properly (which is probably not possible due to missing SDK support now).

**Comment by stephybun** [comment]
@stijndehaes the sdk has been upgraded so it should be possible to implement this now. Are you still interested in working on this?

**Comment by stephybun** [comment]
Closed by #16130
